typing/typecore.ml

```diff
@@ -4764,7 +4764,7 @@ and type_format loc str env =
         mk_exp_loc (Pexp_construct (mk_lid_loc lid, arg)) in
       let mk_cst cst = mk_exp_loc (Pexp_constant cst) in
       let mk_int n = mk_cst (Pconst_integer (Int.to_string n, None))
-      and mk_string str = mk_cst (Pconst_string (str, loc, None))
+      and mk_string str = mk_cst (Pconst_string (str, loc, delim))
```

Comment:
I don't think this is the right way to do it. With this code every call to `mk_string` below (and there are many) will carry the string delimiters. The only string that should carry string delimiters is the one representing the original format string.